backend/controllers/passController.js

```diff
@@ -0,0 +1,90 @@
+const asyncHandler = require("express-async-handler")
+require("dotenv").config()
+const { connectDB } = require("../config/db")
+const User = require("../model/user.model")
+const Pass = require("../model/pass.model")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+
+//@desc Generate a pass by user
+//@route POST /users/generatePass
+const generatePass = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const { name, email, phone, duration, start } = req.body
+
+    const user = await User.findOne({ _id: req.user._id })
+    if (user) {
+      const newPass = await Pass.create({
+        name,
+        phone,
+        email,
+        duration,
+        start,
+        checkInStatus: false,
+        generatedUserId: user.id,
+        userName: req.user.name,
+      })
+      if (newPass) {
+        res.status(201).json({
+          success: true,
+          newPass: newPass,
+          message: "Pass generated successfully",
+        })
+      }
+    } else {
+      throw new ErrorHandler(401, "Signup to create Pass!")
+    }
+  } catch (err) {
+    next(err)
+  }
+})
+
+//@desc View all passes generated by a user
+//@route Get /users/viewPasses
+const viewPasses = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const user = await User.findById(req.user._id)
+
+    const passes = await Pass.find({ generatedUserId: user.id })
+
+    if (passes.length > 0) {
+      res.status(201).json({
+        success: true,
+        passes: passes,
+        message: "Passes fetched successfully",
+      })
+    } else if (passes.length < 1) {
+      throw new ErrorHandler(404, "No passes found")
+    }
+  } catch (err) {
+    next(err)
+  }
+})
+
+//@desc Verify pass by guard
+//@route PUT /users/verifyPass
+const verifyPass = asyncHandler(async (req, res, next) => {
+  try {
+    await connectDB()
+    const user = await User.findById(req.user._id)
+    console.log(req.params._id)
+
+    const passVerified = await Pass.findOneAndUpdate(
+      { _id: req.params._id },
+      {
+        checkInStatus: true,
```

Comment:
Lets continue [this](https://github.com/aditiii23/virtualPermissions/pull/1#discussion_r1055978711) discussion here.

Reply:
yes, if the user has not been permitted into the campus, till then it should be marked as false, and when he wishes to enter then his pass can be marked as true..we can check the pass usage with that variable (initially i was thinking that the pass's checkInStatus could be marked back to false when the person goes out of the campus...)

Reply:
So, the time `checkInstatus` is set to true. The value of checkInTime will change from null to some valid timestamp. Can we interpret that if checkInTime is not null then the user have been entered to the building. This will help us to remove checkInStatus from the db.

Reply:
yes

Reply:
So, please remove it.

Reply:
can we discuss this as well tomorrow?